Cron update (#209)
thijsvanloef/palworld-server-docker: README.md

| BACKUP_ENABLED | Enables automatic backups | true | true/false |
| DELETE_OLD_BACKUPS | Delete backups after a certain number of days                                                                                                                                                       | false          | true/false                                                                                                 |
| OLD_BACKUP_DAYS    | How many days to keep backups                                                                                                                                                                       | 30             | any positive integer                                                                                       |

> | AUTO_UPDATE_CRON_EXPRESSION  | Setting affects frequency of automatic updates. | 0 0 \* \* \* | Needs a Cron-Expression - See [Configuring Automatic Backups with Cron](#configuring-automatic-backups-with-cron) |

Reviewer: Should the default value be daily or hourly? Dockerfile is hourly.

Author: I changed it to hourly in the docs

> | AUTO_UPDATE_ENABLED | Enables automatic updates | true | true/false |

Reviewer: `AUTO_UPDATE_ENABLED` is set to false in the dockerfile.
```suggestion
| AUTO_UPDATE_ENABLED | Enables automatic updates | false | true/false |
```

Author: Yes first wanted to make it default true but since it's not yet tested enough false is better. Forgot to change that.

> | WARN_MINUTES | How long to wait to update the server, after the player were informed. | 30 | !0 |

Reviewer: Need to update variable names here too.

Author: Updated the readme

Set BACKUP_CRON_EXPRESSION to change the default schedule.
Example Usage: If BACKUP_CRON_EXPRESSION to `0 2 * * *`, the backup script will run every day at 2:00 AM.

## Configuring Automatic Updates with Cron

To be able to use automatic Updates with this Server the following environment variables **have** to be set to `true`:
* RCON_ENABLED

> * UPDATE_ON_BOOT

Reviewer: I would make it clear this only works if the container restarts.
You are stopping the game server so the container stops and so long as restart is enabled then it automatically updates otherwise you just stop the container.

It is possible to change it so the container says live and the game is updated. However that would require a pretty big change to either `init.sh` or `start.sh`.

Author: You're right I totally missed that. Thanks

> * RCON_ENABLED
> * UPDATE_ON_BOOT

Reviewer: Should make the linter happy.
```suggestion

* RCON_ENABLED
* UPDATE_ON_BOOT
```

Set BACKUP_CRON_EXPRESSION to change the default schedule.
Example Usage: If BACKUP_CRON_EXPRESSION to `0 2 * * *`, the backup script will run every day at 2:00 AM.

## Configuring Automatic Updates with Cron

To be able to use automatic Updates with this Server the following environment variables **have** to be set to `true`:

* RCON_ENABLED
* UPDATE_ON_BOOT

> you will also need to use the restart policy `always` or `unless-stopped`. The example docker run command and docker compose file in [How to Use](#how-to-use) already use the needed policy

Reviewer: ```suggestion
> [!IMPORTANT]
> 
> If docker restart is not set to policy `always` or `unless-stopped` then the server will shutdown and will need to be
> manually restarted. 
> 
> The example docker run command and docker compose file in [How to Use](#how-to-use) already use the needed policy
```

Set BACKUP_CRON_EXPRESSION to change the default schedule.
Example Usage: If BACKUP_CRON_EXPRESSION to `0 2 * * *`, the backup script will run every day at 2:00 AM.

## Configuring Automatic Updates with Cron

To be able to use automatic Updates with this Server the following environment variables **have** to be set to `true`:

* RCON_ENABLED
* UPDATE_ON_BOOT

> [!IMPORTANT]
> 
> If docker restart is not set to policy `always` or `unless-stopped` then the server will shutdown and will need to be
> manually restarted. 
> 

> > The example docker run command and docker compose file in [How to Use](#how-to-use) already use the needed policy

Author: Np
